common-npm-packages/docker-common-v2/registryauthenticationprovider/registryauthenticationtoken.ts

```diff
@@ -55,13 +55,23 @@ export default class RegistryServerAuthenticationToken {
     }
 }
 
-export function getDockerRegistryEndpointAuthenticationToken(endpointId: string): RegistryServerAuthenticationToken {
+export async function getDockerRegistryEndpointAuthenticationToken(endpointId: string): Promise<RegistryServerAuthenticationToken> {
     var registryType = tl.getEndpointDataParameter(endpointId, "registrytype", true);
     let authToken: RegistryServerAuthenticationToken;
 
     if (registryType === "ACR") {
         const loginServer = tl.getEndpointAuthorizationParameter(endpointId, "loginServer", false).toLowerCase();;
-        authToken = new ACRAuthenticationTokenProvider(endpointId, loginServer).getAuthenticationToken();
+        let authType: string;
+        try {
+            authType = tl.getEndpointAuthorizationScheme(endpointId, false);
+        } catch {
+            authType = tl.getEndpointAuthorizationParameter(endpointId, "scheme", false);
```

Comment:
Can this throw ?

Reply:
This portion of code can throw an error. When I tried to log the changes inside of the ADO pipelines, I couldn't find specifically what error was thrown, but this can throw an error if the scheme is not successfully retrieved.

Reply:
Do you think this should be safe handled then and fallback to SPN ?